TESTS/mbed_platform/FileHandle/TestFile.h

```diff
@@ -0,0 +1,170 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_TESTFILEHANDLE_H
+#define MBED_TESTFILEHANDLE_H
+
+#include "platform/FileHandle.h"
+
+
+#define POS_IS_VALID(pos) (pos >= 0 && pos < _end)
+#define NEW_POS_IS_VALID(pos) (pos >= 0 && pos < (int32_t)FILE_SIZE)
+#define SEEK_POS_IS_VALID(pos) (pos >= 0 && pos <= _end)
+#define FILE_IS_EMPTY() (_end == 0)
+
+template<uint32_t FILE_SIZE>
+class TestFile : public FileHandle {
+public:
+    TestFile(): _pos(0), _end(0) {}
+    ~TestFile() {}
+
+    enum FunctionName {
+        fnNone, fnRead, fnWrite, fnSeek, fnClose, fnIsatty
+    };
+
+    virtual ssize_t read(void *buffer, size_t size)
+    {
+        ssize_t read;
+        _fnCalled = fnRead;
+
+        if(0 == size) {
+            return -EAGAIN;
+        }
+
+        for(read = 0; (size_t)read < size; read++)
+        {
+            if(POS_IS_VALID(_pos)) {
+                ((uint8_t*)buffer)[read] = _data[_pos++];
+            } else {
+                if(0 == read) {
+                    read = -EAGAIN;
```

Comment:
If you're trying to act like a normal file, you should return 0 if _pos >= end of file. As in my previous comment, -EAGAIN is never appropriate for a blocking file.

If you want to test error returns, I'd have some sort of other "set_error" back door control to make it do that instead of the normal operation.

Maybe you could mark a particular byte somewhere in the file invalid, so `if (_pos == _error_pos) indicate EIO` sort of thing, distinct from the "end of file". Dunno.